### PR TITLE
Update code base

### PR DESCRIPTION
**Pull request by paul-crowdbotics**

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "project_slug": "sample",
+    "project_slug": "new-sample",
```
Comment (charathram) [inline cookiecutter.json]: Why is this change needed? Doesn't it get overwritten by our scaffold generator?

File: cookiecutter.json [hunk 2]
```diff
@@ -11,16 +11,16 @@
     "custom_domain": "",
     "top_level_domain_name": "com",
     "second_level_domain_name": "crowdbotics",
-    "project_package_name": "{{ cookiecutter.project_slug|lower|replace(' ', '')|replace('-', '') }}",
-    "database_option": ["h2", "mysql", "mongodb", "postgresql", "none"],
+    "project_package_name": "{{ cookiecutter.project_slug|lower|replace(' ', '')|replace('-', '')|replace('_', '') }}",
+    "database_option": ["mysql", "h2", "mongodb", "postgresql", "none"],
```
Comment (charathram) [inline cookiecutter.json]: Let's leave `h2` as the first (and default) choice.

File: {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}JWTAuthorizationFilter.java{% endif %} [hunk 3]
```diff
@@ -1,62 +1,145 @@
...
+{%- if cookiecutter.has_lombok == "y" %}
+@Slf4j
+{%- endif %}
+public class JWTAuthorizationFilter 
+	extends BasicAuthenticationFilter 
+{
+	/**
+	 * Autowired constructor for {@link JWTAuthorizationFilter}.
+	 *
+	 * @param _authenticationManager		{@link AuthenticationManager}
```
Comment (charathram) [inline {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}JWTAuthorizationFilter.java{% endif %}]: Let's replace tabs between authenticationManager and {@link... with a space

File: {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}SecurityConstants.java{% endif %} [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
+/**
+ * <h1>Security Constants</h1>
+ *
+ * <p>...</p>
+ *
+ * @author crowdbotics.com
+ */
+public class SecurityConstants 
+{
+	public static final String SECRET = "SecretKeyToGenJWTs";
```
Comment (charathram) [inline {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}SecurityConstants.java{% endif %}]: Let's change this secret to `Cr0wdb0t1c515gr3at`

**Review by charathram — CHANGES_REQUESTED** [review]

File: cookiecutter.json (hunk 1 above)
Comment (paul-crowdbotics) [reply cookiecutter.json]: Changed this to not overwrite my old setup. I'll change it back.

File: cookiecutter.json (hunk 2 above)
Comment (paul-crowdbotics) [reply cookiecutter.json]: Done.

File: {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}SecurityConstants.java{% endif %} (hunk 4 above)
Comment (paul-crowdbotics) [reply {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}SecurityConstants.java{% endif %}]: Done

File: {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}JWTAuthorizationFilter.java{% endif %} (hunk 3 above)
Comment (paul-crowdbotics) [reply {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}JWTAuthorizationFilter.java{% endif %}]: Done

Comment (paul-crowdbotics) [reply {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}JWTAuthorizationFilter.java{% endif %}]: Do want the same for JWTAuthenticationFilter.java?

Comment (charathram) [reply {{cookiecutter.snake_case_slug}}/backend/src/main/java/{{cookiecutter.top_level_domain_name}}/{{cookiecutter.second_level_domain_name}}/{{cookiecutter.project_package_name}}/security/{% if cookiecutter.spring_boot_authentication == "jwt" %}JWTAuthorizationFilter.java{% endif %}]: Sure. I see the same in getAuthentication() as well.

**Review by charathram — APPROVED** [review]
